legend-pure-runtime/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/type/_class/ClassImplProcessor.java

```diff
@@ -1081,6 +1061,50 @@ public static String validate(boolean stateAndDeep, CoreInstance _class, String
                                 "        }\n" : "") +
                         "        return " + (stateAndDeep ? "this" : "null") + ";\n" +
                         "    }\n";
+     return validate + validateItems.get();
+
+    }
+
+    private static String validateItem(boolean stateAndDeep, CoreInstance constraint, CoreInstance _class, ProcessorContext processorContext, String extraValues)
+    {
+        ProcessorSupport processorSupport = processorContext.getSupport();
+        SetIterable<? extends CoreInstance> localConstraints = Sets.immutable.withAll(_class.getValueForMetaPropertyToMany(M3Properties.constraints));
+        boolean registerLambdas = localConstraints.contains(constraint);
+        String ruleId = Instance.getValueForMetaPropertyToOneResolved(constraint, M3Properties.name, processorSupport).getName();
+        CoreInstance owner = Instance.getValueForMetaPropertyToOneResolved(constraint, M3Properties.owner, processorSupport);
+        CoreInstance definition = Instance.getValueForMetaPropertyToOneResolved(constraint, M3Properties.functionDefinition, processorSupport);
+        String eval = "(Boolean) " + ValueSpecificationProcessor.createFunctionForLambda(constraint, definition, registerLambdas, processorSupport, processorContext) + ".execute(vars,es)";
+        String messageJavaFunction = null;
+        CoreInstance message = Instance.getValueForMetaPropertyToOneResolved(constraint, M3Properties.messageFunction, processorSupport);
+        if (message != null)
+        {
+            messageJavaFunction = ValueSpecificationProcessor.createFunctionForLambda(constraint, message, registerLambdas, processorSupport, processorContext);
+        }
+
+        if (owner == null || "Global".equals(owner.getName()))
+        {
+            CoreInstance expression = Instance.getValueForMetaPropertyToOneResolved(definition, M3Properties.expressionSequence, processorSupport);
+            CoreInstance constraintClass = Instance.getValueForMetaPropertyToOneResolved(expression, M3Properties.usageContext, processorSupport).getValueForMetaPropertyToOne(M3Properties.type);
+
+            String errorMessage = message == null ?
+                    "\"Constraint :[" + ruleId + "] violated in the Class " + constraintClass.getValueForMetaPropertyToOne(M3Properties.name).getName() + "\"" :
+                    "\"Constraint :[" + ruleId + "] violated in the Class " + constraintClass.getValueForMetaPropertyToOne(M3Properties.name).getName() + ", Message: \" + (String) " + messageJavaFunction + ".execute(vars,es)";
```

Comment:
You need to do string escaping on ruleId and the constraint name to ensure the string is valid Java. You may also need to do the same for messageJavaFunction, though I'm not quite certain what its value will be like.
```suggestion
                    "\"Constraint :[" + StringEscapeUtils.escapeJava(ruleId) + "] violated in the Class " + StringEscapeUtils.escapeJava(constraintClass.getValueForMetaPropertyToOne(M3Properties.name).getName()) + "\"" :
                    "\"Constraint :[" + StringEscapeUtils.escapeJava(ruleId) + "] violated in the Class " + StringEscapeUtils.escapeJava(constraintClass.getValueForMetaPropertyToOne(M3Properties.name).getName()) + ", Message: \" + (String) " + messageJavaFunction + ".execute(vars,es)";
```